src/Assets/Assets.php

```diff
@@ -688,9 +688,16 @@ public function register_in_wp( $assets = null ) {
 		)
 		) {
 			// Registering the asset now would trigger a doing_it_wrong notice: queue the assets to be registered later.
+			if ( $assets === null ) {
+				return;
+			}
 
 			if ( ! is_array( $assets ) ) {
-				$assets = [ $assets ];
+				if ( ! $assets instanceof Asset ) {
+					throw new \InvalidArgumentException( 'Assets in register_in_wp() must be of type Asset' );
+				}
+
+				$assets = [ $assets->get_slug() => $assets ];
```

Comment:
What the check here is doing doesn't seem enough to be sure about what we want to achieve. We might pass an invalid array and the error would still be there and it could be even more serious.

It should be instead:

```php
if ( ! is_array( $assets ) {
    $assets = [ $assets ];
}

foreach ( $assets as $asset {
    if ( ! $asset instanceof Asset ) {
        throw
    }
    
    $this->assets[ $asset->get_slug() ] = $asset;
}

return;
```